scispacy/candidate_generation.py

```diff
@@ -109,39 +131,54 @@ class CandidateGenerator:
     ann_concept_aliases_list: List[str]
         A list of strings, mapping the indices used in the ann_index to possible UMLS mentions.
         This is essentially used a lookup between the ann index and actual mention strings.
-    umls: UmlsKnowledgeBase
+    umls: KnowledgeBase
         A class representing canonical concepts from the Unified Medical Language System knowledge graph.
     verbose: bool
         Setting to true will print extra information about the generated candidates.
     ef_search: int
         The efs search parameter used in the index. This substantially effects runtime speed
         (higher is slower but slightly more accurate). Note that this parameter is ignored
         if a preconstructed ann_index is passed.
-
+    name: str, optional (default = None)
+        The name of the pretrained entity linker to load. Must be one of 'umls' or 'mesh'.
     """
 
     def __init__(
         self,
         ann_index: FloatIndex = None,
         tfidf_vectorizer: TfidfVectorizer = None,
         ann_concept_aliases_list: List[str] = None,
-        umls: UmlsKnowledgeBase = None,
+        umls: KnowledgeBase = None,
         verbose: bool = False,
         ef_search: int = 200,
+        name: str = None,
     ) -> None:
 
+        if name is not None and any(
```

Comment:
This parameter setup seems a bit confusing, but I don't have a suggestion for easy way to fix it so i guess its fine.

Reply:
Yeah, the idea is you _either_ provide a name (which we know about), or you provide all the components yourself.

Reply:
Yeah, the error message is probably good enough, but maybe add a note in the docstring about this point.